File: 프로그래머스/0/120904. 숫자 찾기/숫자 찾기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int num, int k) {
    int answer = 0;
    int cnt = 0;
    int idx = 0;
    
    while(num != 0){
        cnt++;
        if(num % 10 == k){
            idx = cnt;
        }
        num /= 10;
    }
    
    answer = idx != 0 ? cnt - idx + 1 : -1;
    
    return answer;
}

File: 프로그래머스/0/120904. 숫자 찾기/숫자 찾기.java
class Solution {
    public int solution(int num, int k) {
        int answer = -1;
        
        String str = String.valueOf(num);

        for(int i = 0; i < str.length(); i++){
            char c = str.charAt(i);
            if(c == k + '0') {
                answer = i + 1;
                break;
            }
        }
        
        return answer;
    }
}